### ISSUE TITLE
ProblemDetails implementation

### ISSUE THREAD

**Issue by DCichosz** [dj-nitehawk/FastEndpoints#59]
Hi, will you implement ProblemDetails response like it is done in MVC Controller with the Problem() method?

**Comment by dj-nitehawk** [comment]
we already have this functionality but is done a bit differently. example:
```csharp
public override async Task HandleAsync(MyRequest req, CancellationToken ct)
{
    if (req.FullName == null)
        AddError(x => x.FullName, "Full name required!"); //adds an error to the validation errors collection

    if (req.Age <= 5)
        AddError(x => x.Age, "You are too young!");

    ThrowIfAnyErrors(); //sends error/problem response automatically if there are any errors in the collection

    if (DateTime.UtcNow.Date > DateTime.Parse("2022-01-03"))
        ValidationFailures.Add(new("PropertyName", "Too late!")); //add error to the collection

    if (ValidationFailed)
    {
        await SendErrorsAsync(); //sends error response manually.
    }
    else
    {
        await SendOkAsync();
    }
}
```

and if you don't like the shape of the default error response, you can customize it as mentioned here: https://fast-endpoints.com/wiki/Configuration-Settings.html#customizing-error-responses

**Comment by DCichosz** [comment]
I would like to have standard for 4xx responses (example shown here https://codeopinion.com/problem-details-for-better-rest-http-api-errors/).
The SendErrorAsync() forces the 400 code which is not valid in a lot of cases.

I was looking at https://www.nuget.org/packages/Hellang.Middleware.ProblemDetails/ but it's not working with FastEndpoints - only exception handler works.

Is it possible to have something like SendProblemsAsync() which will generate the ProblemDetails object based on RFC 7807 (https://datatracker.ietf.org/doc/html/rfc7807)?

**Comment by stefandevo** [comment]
As @dj-nitehawk pointed out, you can use `ProblemDetails`. I am using this with the standard `Microsoft.AspNetCore.Mvc.ProblemDetails` class; just use the `config.ErrorResponseBuilder` setup to "convert" the standard errors to whatever class you want.

**Comment by DCichosz** [comment]
@stefandevo how do you act on different status codes in the `config.ErrorResponseBuilder` function? Could you show me how you handle creation of ProblemDetails object?

**Comment by dj-nitehawk** [comment]
SendErrorsAsync() now takes an optional status code in `v3.10.0-beta4`.

@DCichosz so you want to send multiple different types of error responses? is that not a kinda nightmare for consumers/clients?

**Comment by DCichosz** [comment]
No. I just want to have one standarized ProblemDetails model that will act on specific status code to generate correct Type and Title of ProblemDetails object.

For 404 status code give me:
Type = https://httpstatuses.com/404
Title = Not Found

 I wanted to use `Microsoft.AspNetCore.WebUtilities.ReasonPhrases` to generate Title based

**Comment by dj-nitehawk** [comment]
so having access to the status code in the `config.ErrorResponseBuilder` func will solve your issue right? 
that way, in the endpoint handler you can do `SendErrorsAsync(4XX)` and in the `ErrorResponseBuilder` you can check the status code and do your transformation to whatever you like.

**Comment by DCichosz** [comment]
That's correct. Then you have the ability to know from where the issue came from. 
You know in the function if the exception Error was sent by FluentValidation (if propertyName is not GeneralErrors and errorCode contains "Validator") or it was your own error that you can handle in any way you want :)

**Comment by dj-nitehawk** [comment]
@DCichosz implemented in `v3.10.0-beta5`
